### PR TITLE
Isolate devnet config, move parsing, dump synchronously

### LINKED ISSUE #218: Dump/load no longer works on 0.2.9
**Describe the bug**
It seems that the state is reset.

**Not reproducible on alpha-goerli**

- [x] This issue is only present on Devnet and cannot be reproduced on alpha-goerli (check the box if true).

**To Reproduce**

1. Start up the devnet using `-dump-path file -dump-on-exit` 
2. Exit 
3. Start up the devnet using `-dump-path file -dump-on-exit --load-path file` 
4. Deploy a wallet using `starknet deploy_account --account test`
5. Check the byte code : `starknet get_code --contract_address [the address]`
6. Exit and restart devnet as in 3.7. 
7. Check again as in 5 -> it's empty.

**Expected behavior**
In 0.2.8 and earlier, the byte code is returned, indicating the state is lost.

**Devnet version**

Work on 0.2.6, 0.2.7, 0.2.8
Broken on 0.2.9

**Machine specifications**

- OS: Mac 12.3.1
- CPU: M1 arm
- Memory: 64G

### PR DESCRIPTION
**Pull request by FabijanC**
## Usage related changes

- Fix dump/load and restart functionalities (fix #218)
- Wait for dump to finish (potentially a slowdown, but should eliminate race-related bugs)

## Development related changes

- Predeploy 1 account by default in tests (instead of 0 - this caused headache while trying to figure out why tests were passing but real-world was failing)
- Make rpc tests more independent of each other:
  - Set rpc fixtures module to function instead of scope
  - Restart state after each test
- Refactor code:
  - Expand functionality of `DevnetConfig` and move it from `starknet_wrapper.py` to `devnet_config.py`
  - Move `parse_args` from `util.py` to `devnet_config.py`
  - Moved account generation from `server.py` to `__init__` of `Accounts`.

## Checklist:

- [x] No linter errors
- [x] Performed a self-review of the code
- [x] Rebased to the last commit of the target branch (or merged it into my branch)
- [x] Documented the changes
- [x] Linked the issues which this PR resolves
- [x] Updated the tests
- [x] All tests are passing

### CODE REVIEW

File: starknet_devnet/state.py [hunk 1]
```diff
@@ -2,30 +2,30 @@
...
-        self.dumper = Dumper(self.starknet_wrapper)
+        self.set_starknet_wrapper(StarknetWrapper(DevnetConfig()))
 
-    def __set_starknet_wrapper(self, starknet_wrapper: StarknetWrapper):
+    def set_starknet_wrapper(self, starknet_wrapper: StarknetWrapper):
         """Sets starknet wrapper and creates new instance of dumper"""
         self.starknet_wrapper = starknet_wrapper
         self.dumper = Dumper(starknet_wrapper)
 
-    async def reset(self, config: DevnetConfig = None):
```
Comment (FabijanC) [inline starknet_devnet/state.py:21]: This parameter was never used.

File: starknet_devnet/state.py [hunk 2]
```diff
@@ -2,30 +2,30 @@
...
     """
     Stores starknet wrapper and dumper
     """
     def __init__(self):
-        self.starknet_wrapper = StarknetWrapper(config=DevnetConfig())
-        self.dumper = Dumper(self.starknet_wrapper)
+        self.set_starknet_wrapper(StarknetWrapper(DevnetConfig()))
 
-    def __set_starknet_wrapper(self, starknet_wrapper: StarknetWrapper):
+    def set_starknet_wrapper(self, starknet_wrapper: StarknetWrapper):
```
Comment (FabijanC) [inline starknet_devnet/state.py:16]: Made public so it can be used from `main` in `server.py`.

File: starknet_devnet/starknet_wrapper.py [hunk 3]
```diff
@@ -71,6 +65,11 @@ def __init__(self, config: DevnetConfig):
         self.fee_token = FeeToken(self)
         self.accounts = Accounts(self)
 
+        if config.start_time is not None:
```
Comment (FabijanC) [inline starknet_devnet/starknet_wrapper.py:68]: Moved here from server.py

File: starknet_devnet/server.py [hunk 4]
```diff
@@ -123,22 +92,17 @@ def load_config(self):
...
 
-    args = parse_args()
-
     # Uncomment this once fork support is added
     # origin = Origin(args.fork) if args.fork else NullOrigin()
     # starknet_wrapper.origin = origin
 
+    args = parse_args(sys.argv[1:])
+    state.set_starknet_wrapper(StarknetWrapper(DevnetConfig(args)))
     load_dumped(args)
```
Comment (tabaktoni) [inline starknet_devnet/server.py:104]: what do you think about?
state.dumper.load_dumped(args)
state.dumper.set_dump_options(args)

It implicates move dumper functions from server to dump.py

File: starknet_devnet/server.py [hunk 5]
```diff
@@ -123,22 +92,17 @@ def load_config(self):
...
     """Runs the server."""
 
-    args = parse_args()
-
     # Uncomment this once fork support is added
     # origin = Origin(args.fork) if args.fork else NullOrigin()
     # starknet_wrapper.origin = origin
 
+    args = parse_args(sys.argv[1:])
+    state.set_starknet_wrapper(StarknetWrapper(DevnetConfig(args)))
```
Comment (tabaktoni) [inline starknet_devnet/server.py:103]: It is called two times when you start an app on state init and after that here again.

**Review by tabaktoni — APPROVED** [review]
Like this direction in separation of concerns

File: starknet_devnet/server.py (hunk 5 above)
Comment (FabijanC) [reply starknet_devnet/server.py]: Doesn't affect performance, tests using app.test_client don't work without it.

File: starknet_devnet/server.py (hunk 4 above)
Comment (FabijanC) [reply starknet_devnet/server.py]: https://github.com/Shard-Labs/starknet-devnet/pull/219/commits/95372005781725c71157cd5ac74ce9e644e9542b

**Pull request merged**